### PR TITLE
check if modified files after update_config/supported

### LINKED ISSUE #661: [FEA] CI premerge check should fail if build modifies files in the repository
**Is your feature request related to a problem? Please describe.**
The `verify` build automatically updates `docs/configs.md` if necessary, based on the contents of `RapidsConf`.  Sometimes a PR may be submitted that should have modified generated documentation, but it was missed somehow.  The CI premerge check should catch this and flag the PR accordingly.

**Describe the solution you'd like**
At least as it is now, any build that starts with no modified repository files should complete without modifying repository files.  The CI premerge check should look for modified repository files after a build (e.g.: `git status -s | grep '^ M'`), and if any are found, it should flag an error during the premerge check.

### PR DESCRIPTION
**Pull request by pxLi**
Signed-off-by: Peixin Li <pxli@nyu.edu>

fix #661

validated working inside this PR w/ different commits

### CODE REVIEW

**Comment by pxLi** [comment]
build

**Comment by pxLi** [comment]
build

**Comment by pxLi** [comment]
build

File: sql-plugin/src/main/scala/com/nvidia/spark/rapids/RapidsConf.scala [hunk 1]
```diff
@@ -298,7 +298,7 @@ object RapidsConf {
 
   // Resource Configuration
 
-  val PINNED_POOL_SIZE = conf("spark.rapids.memory.pinnedPool.size")
+  val PINNED_POOL_SIZE = conf("spark.rapids.memory.pinnedPool.size-DUMMY_MODIFIED")
```
Comment (pxLi) [inline sql-plugin/src/main/scala/com/nvidia/spark/rapids/RapidsConf.scala]: this one failed (change to RpiadConf only but not config.md) as expected [build 1023](https://blossom.nvidia.com/sw-gpu-spark-jenkins/job/rapids_premerge-github/1023/console)
image

**Comment by pxLi** [comment]
CI Passed after remove the DUMMY change

**Review by jlowe — APPROVED** [review]

**Pull request merged**